plugins/inventory/aws_ec2.py

```diff
@@ -32,16 +32,17 @@
   hostnames:
     description:
       - A list in order of precedence for hostname variables.
+      - The elements of the list can be a dict with the keys mentioned below or a string.
+      - Can be one of the options specified in U(http://docs.aws.amazon.com/cli/latest/reference/ec2/describe-instances.html#options).
+      - If value provided does not exist in the above options, it will be used as a literal string.
+      - To use tags as hostnames use the syntax tag:Name=Value to use the hostname Name_Value, or tag:Name to use the value of the Name tag.
     type: list
-    elements: dict
+    elements: raw
```

Comment:
```suggestion
    elements: raw
```

While the documentation only mentions dict and string, there's the risk someone would pass something weird and get type errors instead of a "friendly" error message.  If you're dropping the type checking here, then there should be type checking performed in the code.